src/core/public/workspace/workspaces_client.ts

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { resolve as resolveUrl } from 'url';
+import type { PublicMethodsOf } from '@osd/utility-types';
+import { WORKSPACES_API_BASE_URL } from '../../server/types';
+import { HttpStart } from '../http';
+import { WorkspaceAttribute, WorkspaceFindOptions } from '.';
+
+/**
+ * WorkspacesClientContract as implemented by the {@link WorkspacesClient}
+ *
+ * @public
+ */
+export type WorkspacesClientContract = PublicMethodsOf<WorkspacesClient>;
+
+const join = (...uriComponents: Array<string | undefined>) =>
+  uriComponents
+    .filter((comp): comp is string => Boolean(comp))
+    .map(encodeURIComponent)
+    .join('/');
+
+type IResponse<T> =
+  | {
+      result: T;
+      success: true;
+    }
+  | {
+      success: false;
+      error?: string;
+    };
+
+/**
+ * Workspaces is OpenSearchDashboards's visualize mechanism allowing admins to
+ * organize related features
+ *
+ * @public
+ */
+export class WorkspacesClient {
+  private http: HttpStart;
+  constructor(http: HttpStart) {
+    this.http = http;
+  }
+
+  private getPath(path: Array<string | undefined>): string {
+    return resolveUrl(`${WORKSPACES_API_BASE_URL}/`, join(...path));
+  }
+
+  public async enterWorkspace(id: string): Promise<IResponse<null>> {
+    return {
+      success: false,
+      error: 'Unimplement',
+    };
+  }
+
+  public async exitWorkspace(): Promise<IResponse<null>> {
+    return {
+      success: false,
+      error: 'Unimplement',
+    };
+  }
+
+  public async getCurrentWorkspaceId(): Promise<IResponse<WorkspaceAttribute['id']>> {
```

Comment:
@SuZhou-Joe Can response of this function not be a promise?

Reply:
It will be more extensible that we claim the response as promise  and it may require some API calls when we need to fetch the id.

Reply:
In my opinion, the service should have a state to store current workspace info and have a synchronous method to return so that other component can handle this logic easily. We can discuss this in scrum meeting.